src/web/js-interop.md

```diff
@@ -22,3 +22,77 @@ For help using the `js` package, see the following:
 [js-api]: {{site.pub-api}}/js
 [sass]: {{site.pub-pkg}}/sass
 
+## Future-proof JS interop
+
+Dart's JS interop story is currently evolving. 
+Many of the features that enable future JS interop,
+and support of all three compilers,
+are ready to use (Dart version 2.19). 
+To ease the transition to future JS interop,
+you can refactor your code to conform to the
+syntax and semantics of the new features now.
+
+{{site.alert.note}}
+  Future-proofing is for those *very* eager to
+  prepare their code for [Dart 3][], or to try
+  integrating with [`dart2wasm`][] before Dart 3.
+  None of this necessarily works *performantly* on
+  `dart2wasm` yet, but refactoring now means you
+  won't need to later, if that's important to you.
```

Comment:
@srujzs – don't we get a bit of size/speed improvement using the new stuff NOW?

Reply:
I seemed to get a little one in https://github.com/dart-lang/dartdoc/pull/3299 :)

Reply:
Similar note as I mentioned above: refactoring now means simpler refactoring later, but we can't promise no changes later.

Re performance, the current statement seems OK to me:
* dart2wasm has known overheads that are not polished yet (e.g. cost of a trampoline between JS and wasm is high right now)
* dart2js performance can be observably better or worse depending on the scenario. Calls are more direct, code-size may be smaller, but there are new soundness checks at the boundary that could potentially make some APIs slower. 

Circling back to the intent of this note, I'd say this preview of the next gen interop is meant mostly for experimentation: `package:js` interop as it exist today (w/o `@staticInterop`) will not be supported by dart2wasm, `dart:js_util` is supported, but it's not as ergonomic and will not be optimized. So we expect that anyone interested in trying dart2wasm that also needs jsinterop, will likely be interested in this newer form of JSInterop, even though more changes are expected as we introduce inline-classes and make additional tweaks in the next few months.

Reply:
I can remove that clause and just say "Refactoring now means...."